### ISSUE TITLE
[Bug]: the perf is much lower than mysql  for batch insert of tpcc data

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#5530]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.6.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

run load data of 1 warehouse to mo through benchbase tools, but the cost is very different.

mysql is about 80s，but mo is more than 1 hour

**Comment by daviszhen** [comment]
provide some cases

**Comment by florashi181** [comment]
first, profiling it ~ @jianwan0214

**Comment by YANGGMM** [comment]
Still working on it.

**Comment by YANGGMM** [comment]
Have finished profiling with the help of @jianwan0214  as  
![profile](https://user-images.githubusercontent.com/26563383/200552521-7f8f7aa3-4aad-4fb1-90bc-156bf3d11ce8.png)

**Comment by florashi181** [comment]
so what's the action or suggestion next? @YANGGMM @jianwan0214

**Comment by fengttt** [comment]
Please ping @LeftHandCold to see if this is the logging issue.   @xzxiong may have fixed it.

**Comment by YANGGMM** [comment]
> Please ping @LeftHandCold to see if this is the logging issue. @xzxiong may have fixed it.

ok

**Comment by xzxiong** [comment]
brief: 
1. after running load  2 min, gc cost much
2. after running laod 12 min, txnbase cost much

-----

dev:  dev1
run: `./mo-service -debug-http 127.0.0.1:8123 -allocs-profile ./allocs.profile -launch ./etc/launch-tae-logservice/launch.toml`
test case:
1. clone https://github.com/aressu1985/mo-tpcc
3. startup mo
4. init schema
```
mysql> create database tpcc;
mysql>  use tpcc;
mysql> source /path/to/mo-tpcc/sql/tableCreates_pk.sql
```
5.  exec load
```
$ cd /path/to/mo-tpcc
$ mkdir data
$ sh ./runLoader.sh props.mo warehouse 1
```
6. gen cpu profile `PPROF_TMPDIR="./tmpdir" go tool pprof http://localhost:8123/debug/pprof/profile`
- after running load 2 minutes
[pprof.mo-service.samples.cpu.001.pb.gz](https://github.com/matrixorigin/matrixone/files/9972332/pprof.mo-service.samples.cpu.001.pb.gz)

<img width="1525" alt="image" src="https://user-images.githubusercontent.com/3927687/200867178-a6ad2f8f-7189-42c6-a5ab-894f7996234e.png">

![profile001](https://user-images.githubusercontent.com/3927687/200867850-91429eb5-9e6a-4385-a13b-a7d7b0f42651.svg)



- after running 12 minutes
[pprof.mo-service.samples.cpu.002.pb.gz](https://github.com/matrixorigin/matrixone/files/9972350/pprof.mo-service.samples.cpu.002.pb.gz)

<img width="1533" alt="image" src="https://user-images.githubusercontent.com/3927687/200867260-96b1e82b-e9bc-4fe8-b8cd-d6f2646b9f12.png">

![profile002](https://user-images.githubusercontent.com/3927687/200867715-0d3a0bf1-ba76-4cf4-8bbf-70de4f2ac702.svg)

**Comment by xzxiong** [comment]
load run 26 min
```
[2022-11-09 22:43:16] init load
mkdir: cannot create directory 'data': File exists
Starting BenchmarkSQL LoadData

props.mo
driver=com.mysql.cj.jdbc.Driver
conn=jdbc:mysql://127.0.0.1:6001/tpcc?characterSetResults=utf8&continueBatchOnError=false&useServerPrepStmts=true&alwaysSendSetIsolation=false&useLocalSessionState=true&zeroDateTimeBehavior=CONVERT_TO_NULL&failoverReadOnly=false&serverTimezone=Asia/Shanghai&enabledTLSProtocols=TLSv1.2&useSSL=false
user=dump
password=***********
warehouses=1
loadWorkers=4
fileLocation (not defined)
csvNullValue (not defined - using default '')

Worker 000: Loading ITEM
Worker 001: Loading Warehouse      1
Worker 000: Loading ITEM done
Worker 001: Loading Warehouse      1 done
[2022-11-09 23:09:28] done load
```

**Comment by YANGGMM** [comment]
@LeftHandCold  plz have a look.

**Comment by LeftHandCold** [comment]
I will debug this issue in the next two days

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
@LeftHandCold How's it going？

**Comment by YANGGMM** [comment]
still working on it.

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by YANGGMM** [comment]
still working on it

**Comment by jiangxinmeng1** [comment]
In CN-DN, memtable.(*Version[...]).Visible cost much
![微信图片_20220409181135](https://user-images.githubusercontent.com/51114574/208590319-80f13570-7c99-4253-8f49-e362c0fadcd7.jpg)

**Comment by XuPeng-SH** [comment]
> brief:
> 
> 1. after running load  2 min, gc cost much
> 2. after running laod 12 min, txnbase cost much
> 
> dev: dev1 run: `./mo-service -debug-http 127.0.0.1:8123 -allocs-profile ./allocs.profile -launch ./etc/launch-tae-logservice/launch.toml` test case:
> 
> 1. clone https://github.com/aressu1985/mo-tpcc
> 2. startup mo
> 3. init schema
> 
> ```
> mysql> create database tpcc;
> mysql>  use tpcc;
> mysql> source /path/to/mo-tpcc/sql/tableCreates_pk.sql
> ```
> 
> 5. exec load
> 
> ```
> $ cd /path/to/mo-tpcc
> $ mkdir data
> $ sh ./runLoader.sh props.mo warehouse 1
> ```
> 
> 6. gen cpu profile `PPROF_TMPDIR="./tmpdir" go tool pprof http://localhost:8123/debug/pprof/profile`
> 
> * after running load 2 minutes
>   [pprof.mo-service.samples.cpu.001.pb.gz](https://github.com/matrixorigin/matrixone/files/9972332/pprof.mo-service.samples.cpu.001.pb.gz)
> 
> <img alt="image" width="1525" src="https://user-images.githubusercontent.com/3927687/200867178-a6ad2f8f-7189-42c6-a5ab-894f7996234e.png">
> 
> ![profile001](https://user-images.githubusercontent.com/3927687/200867850-91429eb5-9e6a-4385-a13b-a7d7b0f42651.svg)
> 
> * after running 12 minutes
>   [pprof.mo-service.samples.cpu.002.pb.gz](https://github.com/matrixorigin/matrixone/files/9972350/pprof.mo-service.samples.cpu.002.pb.gz)
> 
> <img alt="image" width="1533" src="https://user-images.githubusercontent.com/3927687/200867260-96b1e82b-e9bc-4fe8-b8cd-d6f2646b9f12.png">
> 
> ![profile002](https://user-images.githubusercontent.com/3927687/200867715-0d3a0bf1-ba76-4cf4-8bbf-70de4f2ac702.svg)

We're investigating CN-DN deployment and what you post is a standalone deployment.

**Comment by reusee** [comment]
I did the following:

start mo-service with: 
```
./mo-service -launch etc/launch-tae-CN-tae-DN/launch.toml -debug-http :9876 2>&1 | tee -i log.txt
```

load schema with:
```
create database tpcc;
use tpcc;
source /Users/reus/matrixorigin/mo-tpcc/sql/tableCreates.sql;
```

load data with:
```
./runLoader.sh props.mo warehouse 1
```

during the load, fetch the CPU profile with:
```
 curl 'localhost:9876/debug/pprof/profile?seconds=120' > prof
```

open the profile with:
```
go tool pprof -http :6060 prof
```

**Comment by reusee** [comment]
I'll try to replace the old mem table with the new implementation in the CN partition cache.

**Comment by reusee** [comment]
in progress

**Comment by reusee** [comment]
in progress

**Comment by reusee** [comment]
in progress

**Comment by reusee** [comment]
in progress

**Comment by reusee** [comment]
in progress

**Comment by reusee** [comment]
on sick leave.

**Comment by reusee** [comment]
on sick leave.

**Comment by reusee** [comment]
in progress

**Comment by reusee** [comment]
in progress

**Comment by nnsgmsone** [comment]
空了开始研究

**Comment by nnsgmsone** [comment]
insert with auto is too heavy, need to find ways to reduce overhead.

**Comment by fengttt** [comment]
We believe https://github.com/matrixorigin/matrixone/issues/6956 should resolve this issue.

**Comment by ouyuanning** [comment]
not work on it today

**Comment by ouyuanning** [comment]
not work on it today

**Comment by fengttt** [comment]
@ouyuanning this is a S-1.

**Comment by ouyuanning** [comment]
sorry , not to comment clearly: 
@iceTTTT  have a pr https://github.com/matrixorigin/matrixone/pull/7659 to do this。
I have ask him to make a meeting to discuss the code

**Comment by ouyuanning** [comment]
https://github.com/matrixorigin/matrixone/pull/7659  was on reviewing

**Comment by ouyuanning** [comment]
https://github.com/matrixorigin/matrixone/pull/7659 on reviewing

**Comment by ouyuanning** [comment]
we have auto incr cache now.
pls confirm that.

**Comment by aressu1985** [comment]
till now, cache does not work

**Comment by iceTTTT** [comment]
Made a few changes in cache, may make difference.

**Comment by ouyuanning** [comment]
Need confirm

**Comment by iceTTTT** [comment]
Test locally. Inserting **10w** rows to table described below:

 t1(a int auto_increment, b int unique key, c varchar(20), d decimal(7,2))

![image](https://user-images.githubusercontent.com/74845916/218427080-f0fbdf35-a03a-4d25-9fff-dd0fa3695aaa.png)

And it costs about **20min**.

**Comment by ouyuanning** [comment]
init MO like:
```
select mo_ctl('cn', 'task', 'disable'); 
create database db1;
use db1;
create table t1(a int, b int, c varchar(20), d decimal(7,2));
```

run insert with jdbc. code like:
```
package io.mo;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class App 
{
    public static void main( String[] args )
    {  
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            String url = "jdbc:mysql://127.0.0.1:6001/db1?characterSetResults=utf8&continueBatchOnError=false&useServerPrepStmts=true&alwaysSendSetIsolation=false&useLocalSessionState=true&zeroDateTimeBehavior=CONVERT_TO_NULL&failoverReadOnly=false&serverTimezone=Asia/Shanghai";
            Connection con = DriverManager.getConnection(url,"dump","111");
            long startTime=System.currentTimeMillis();

            int batchCount = 2000;
            PreparedStatement stmt = con.prepareStatement("insert into t1(a,b,c,d) values (?, ?, ?, ?)");
            for (int i =0; i< batchCount; i ++) {                
                stmt.setInt(1,i);
                stmt.setInt(2,i);
                stmt.setString(3,"aaaaa");
                stmt.setFloat(4,(float) 12.20);
                stmt.execute();
            }
            con.close();
            long endTime=System.currentTimeMillis();
            System.out.println("Insert " + (batchCount) + "rows in "+(endTime-startTime) / 1000 +"s");
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}
```

java program get result:
```
Insert 2000rows in 18s
```

prof cmd in ：
```
go tool pprof http://localhost:6060/debug/pprof/profile?seconds=10
```

get result:
[pprof.samples.cpu.017.pb.gz](https://github.com/matrixorigin/matrixone/files/10730252/pprof.samples.cpu.017.pb.gz)

png file:
![profile001](https://user-images.githubusercontent.com/45346669/218677738-3ef27a13-5bcc-4f79-adac-9ae9eba93209.png)

**Comment by ouyuanning** [comment]
i had test table like :
```
create table t1(a int primary key auto_increment, b int unique key, c varchar(20), d decimal(7,2));
```
 the result is not different. MO also need 19/20 seconds to insert 2000 rows.

and i had test java code to use executeBatch.  java code like
```
package io.mo;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class App 
{
    public static void main( String[] args )
    {  
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            String url = "jdbc:mysql://127.0.0.1:6001/db1?characterSetResults=utf8&continueBatchOnError=false&useServerPrepStmts=true&alwaysSendSetIsolation=false&useLocalSessionState=true&zeroDateTimeBehavior=CONVERT_TO_NULL&failoverReadOnly=false&serverTimezone=Asia/Shanghai";
            Connection con = DriverManager.getConnection(url,"dump","111");
            long startTime=System.currentTimeMillis();

            int insertNum = 100;
            int totolInsertCount = 0;
            int batchCount = 20;

            PreparedStatement stmt = con.prepareStatement("insert into t1(a,b,c,d) values (?, ?, ?, ?)");
            for (int i =0; i< batchCount; i ++) {
    
                for (int j =0; j<insertNum; j++) {
                    stmt.setInt(1,i*insertNum+j);
                    stmt.setInt(2,i*insertNum+j);
                    stmt.setString(3,"aaaaa");
                    stmt.setFloat(4,(float) 12.20);
                    stmt.addBatch();
                }
                
                stmt.executeBatch();
                stmt.clearBatch();
                totolInsertCount = totolInsertCount + insertNum;
                long endTime=System.currentTimeMillis();
                System.out.println("Insert " + (totolInsertCount) + "rows in "+(endTime-startTime) / 1000 +"s");
            }
            con.close();
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}
```

run result is 
```
Insert 100rows in 0s
Insert 200rows in 1s
Insert 300rows in 2s
Insert 400rows in 3s
Insert 500rows in 4s
Insert 600rows in 5s
Insert 700rows in 6s
Insert 800rows in 7s
Insert 900rows in 8s
Insert 1000rows in 8s
Insert 1100rows in 9s
Insert 1200rows in 10s
Insert 1300rows in 11s
Insert 1400rows in 12s
Insert 1500rows in 13s
Insert 1600rows in 14s
Insert 1700rows in 15s
Insert 1800rows in 16s
Insert 1900rows in 17s
Insert 2000rows in 17s
```

**Comment by ouyuanning** [comment]
i think that 's not auto incr or unique key's problem.
after discuss with @cnutshell . we see lot's go routine running & lot's routine wait.

**Comment by ouyuanning** [comment]
working on https://github.com/matrixorigin/matrixone/issues/6567.

**Comment by ouyuanning** [comment]
working on https://github.com/matrixorigin/matrixone/issues/6567.

**Comment by ouyuanning** [comment]
wroking on https://github.com/matrixorigin/matrixone/issues/5476

**Comment by ouyuanning** [comment]
after trace & prof. I think I have found the key make insert so slower.
1、test base env:
we have schema in MO:
```
create database db1;
use db1;
create table t1(a int, b int);
```

2、we use a jave jdbc client to insert rows to MO.
java code like:
```
select mo_ctl('cn', 'task', 'disable');
create database db1;
use db1;
create table t1(a int, b int);


package io.mo;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class App 
{
    public static void main( String[] args )
    {  
        try {
            Class.forName("com.mysql.cj.jdbc.Driver");
            String url = "jdbc:mysql://127.0.0.1:6001/db1?characterSetResults=utf8&continueBatchOnError=false&useServerPrepStmts=true&alwaysSendSetIsolation=false&useLocalSessionState=true&zeroDateTimeBehavior=CONVERT_TO_NULL&failoverReadOnly=false&serverTimezone=Asia/Shanghai";
            Connection con = DriverManager.getConnection(url,"dump","111");
            long startTime=System.currentTimeMillis();

            int insertNum = 100;
            int totolInsertCount = 0;
            int batchCount = 20;

            PreparedStatement stmt = con.prepareStatement("insert into t1(a,b) values (?, ?)");
            for (int i =0; i< batchCount; i ++) {
    
                for (int j =0; j<insertNum; j++) {
                    stmt.setInt(1,i*insertNum+j);
                    stmt.setInt(2,i*insertNum+j);
                    stmt.addBatch();
                }
                
                stmt.executeBatch();
                stmt.clearBatch();
                totolInsertCount = totolInsertCount + insertNum;
                long endTime=System.currentTimeMillis();
                System.out.println("Insert " + (totolInsertCount) + "rows in "+(endTime-startTime) / 1000 +"s");
            }
            con.close();
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}
```

and this client take 17s to finish the insert action(20 * 100 = 2000 rows)

3、I found that, we have two problem to make insert slower 
    3.1、dragonboat's problem:
        method sequentialSaveState of LogDB (in dragonboat/internal/tan/logdb.go), that will sync after db.write, that use lot's time and block the go routine.

        i try to change the code from:
    ```
        sync, err := db.write(ud, buf)
        if err != nil {
            return err
        }
        if sync {
            wg.Add(1)
            go func() {
                if err := db.sync(); err != nil {
                    panicNow(err)
                }
                wg.Done()
            }()
        }
    ```

    to
    ```
        _, err = db.write(ud, buf)
        if err != nil {
            return err
        }
        //if sync {
        //    wg.Add(1)
        //    go func() {
        //        if err := db.sync(); err != nil {
        //            panicNow(err)
        //        }
        //        wg.Done()
        //    }()
        //}
    ```

    the java client will take 5s(from 17s to 5s) to finish the insert action（20 * 100 = 2000 rows)

      
    3.2、MO's problem
        we need to check user have priv to insert in that table(t1). that take lot's time.
        if I change determineUserHasPrivilegeSet (in  mo/pkg/frontend/authenticate.go). make it always do nothing but return true, nil;
        the java client will take 3s（from 5s to 3s） to finish the insert action（20 * 100 = 2000 rows)

**Comment by ouyuanning** [comment]
I think the third problem is logUtil
if i change log level from "info" to "error"
the java client will take 2s（from 3s to 2s） to finish the insert action（20 * 100 = 2000 rows)

**Comment by fengttt** [comment]
@reusee run a query to find out how many log lines are written.    See if we are too verbose.

**Comment by fengttt** [comment]
2000 lines should take, well, 0.1 seconds?

**Comment by nnsgmsone** [comment]
waiting for process

**Comment by nnsgmsone** [comment]
waiting for process

**Comment by aressu1985** [comment]
for commit : 845d784c7fa143fa8ac39eb5b1ded16ab96b2218
After executing the './runLoader.sh props.mo warehouse 1'  on mo-srv-128, for about 11 minutes, the mo return error:

java.sql.SQLException: internal error: message body 329104195 is too large, max is 104857600
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
	at com.mysql.cj.jdbc.ConnectionImpl.commit(ConnectionImpl.java:813)
	at io.mo.LoadDataWorker.loadWarehouse(LoadDataWorker.java:765)
	at io.mo.LoadDataWorker.run(LoadDataWorker.java:196)
	at java.lang.Thread.run(Thread.java:748)
Worker 001: ERROR: internal error: message body 329104195 is too large, max is 104857600

It is the same as #7767 


for mysql,it only cost 33 s:

[sudong@mo-srv-128 mo-tpcc]$ time ./runLoader.sh props.mysql 1 warehouse
Starting BenchmarkSQL LoadData

props.mysql
driver=com.mysql.cj.jdbc.Driver
conn=jdbc:mysql://127.0.0.1:3306/tpcc?characterSetResults=utf8&continueBatchOnError=false&useServerPrepStmts=true&alwaysSendSetIsolation=false&useLocalSessionState=true&zeroDateTimeBehavior=CONVERT_TO_NULL&failoverReadOnly=false&serverTimezone=Asia/Shanghai&enabledTLSProtocols=TLSv1.2&useSSL=false
user=dump
password=***********
warehouses=1
loadWorkers=4
fileLocation (not defined)
csvNullValue (not defined - using default '')

Worker 000: Loading ITEM
Worker 001: Loading Warehouse      1
Worker 000: Loading ITEM done
Worker 001: Loading Warehouse      1 done

real	0m33.054s
user	0m11.015s
sys	0m4.612s

**Comment by nnsgmsone** [comment]
I will deal with the memory issue first

**Comment by nnsgmsone** [comment]
I will deal with the memory issue first

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
等处理完https://github.com/matrixorigin/matrixone/issues/7891在处理这个。

**Comment by nnsgmsone** [comment]
This case is where two transactions exist, and then each transaction is inserted line by line. I'm not sure if this makes sense for tpcc. @aressu1985

**Comment by nnsgmsone** [comment]
还在定位，这是一些trace的分析情况。目前的问题是要找出那些大的延迟是哪里导致的:

![截图 2023-04-23 10-01-09](https://user-images.githubusercontent.com/31609524/233815498-828e76a6-f668-4257-a2f5-6cb5b044d383.png)

![截图 2023-04-23 10-01-20](https://user-images.githubusercontent.com/31609524/233815503-63ec5f7e-f6e5-4330-a578-f477841fe982.png)

**Comment by nnsgmsone** [comment]
看trace记录，比如构建流程，plan.Buildplan和pipeline.Compile都不会存在耗时很久的情况可是TxnComputataionWrapper.Compile存在耗时25ms的情况。接下来定位原因。

**Comment by nnsgmsone** [comment]
![截图 2023-04-23 10-15-18](https://user-images.githubusercontent.com/31609524/233815825-69d19948-db46-4798-8229-c7e9f1061226.png)
权限验证应该是需要优化的第一个地方。

**Comment by xzxiong** [comment]
@nnsgmsone  `Compile` and `authenticateCanExecuteStatementAndPlan` 两者都需要 `open new txn`, 在并发大时耗时会很大
这里有一个例子：耗时 5s 的查询，在这两处分别花费了 3.7 s 和 1.2 s
https://github.com/matrixorigin/matrixone/issues/8786#issuecomment-1514600090

**Comment by nnsgmsone** [comment]
我统计了new transcation和commit的开销的，我也一起贴一下. @xzxiong 
![截图 2023-04-23 10-42-29](https://user-images.githubusercontent.com/31609524/233816693-da2432be-5ab4-48e7-b42a-982e837353ad.png)
![截图 2023-04-23 10-42-42](https://user-images.githubusercontent.com/31609524/233816702-18129a84-4347-4b59-957f-5cdffa0e7565.png)

**Comment by nnsgmsone** [comment]
![截图 2023-04-23 10-42-16](https://user-images.githubusercontent.com/31609524/233816735-6190916f-5c51-4dc3-a1d0-46f1a1f2f62c.png)

**Comment by nnsgmsone** [comment]
Found that prepare did not work well and continued to track

**Comment by nnsgmsone** [comment]
continue work

**Comment by nnsgmsone** [comment]
continue work

**Comment by nnsgmsone** [comment]
目前的prepare实现对于insert不够友好，先处理一下#643后修改一些prepare的实现

**Comment by nnsgmsone** [comment]
先帮忙远宁处理plan分拆的问题

**Comment by nnsgmsone** [comment]
先帮忙远宁处理plan分拆的问题

**Comment by nnsgmsone** [comment]
fixed by #9372

**Comment by aressu1985** [comment]
testing

**Comment by aressu1985** [comment]
fixed:2d27fc646c470534d53f235f9aff397599b4fe0b

10 warehouse:
[sudong@VM-16-14-centos mo-tpcc]$ time ./runLoader.sh props.mo warehouse 10
Starting BenchmarkSQL LoadData

props.mo
driver=com.mysql.cj.jdbc.Driver
conn=jdbc:mysql://127.0.0.1:6001/tpcc?characterSetResults=utf8&continueBatchOnError=false&useServerPrepStmts=true&alwaysSendSetIsolation=false&useLocalSessionState=true&zeroDateTimeBehavior=CONVERT_TO_NULL&failoverReadOnly=false&serverTimezone=Asia/Shanghai&useSSL=false&socketTimeout=30000
user=dump
password=***********
warehouses=1
loadWorkers=4
fileLocation (not defined)
csvNullValue (not defined - using default '')

Worker 000: Loading ITEM
Worker 001: Loading Warehouse      1
Worker 000: Loading ITEM done
Worker 001: Loading Warehouse      1 done

real	2m49.171s
user	0m9.413s
sys	0m5.942s

1 warehouse:
